bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/wizard/NewIDFProjectWizard.java

```diff
@@ -114,30 +115,44 @@ public boolean performFinish()
 		}
 
 		final String target = projectCreationWizardPage.getSelectedTarget();
-		this.getShell().addDisposeListener(new DisposeListener()
-		{
-			@Override
-			public void widgetDisposed(DisposeEvent event)
+		this.getShell().addDisposeListener(event -> {
+			ILaunchBarManager launchBarManager = UIPlugin.getService(ILaunchBarManager.class);
+			TargetSwitchJob targetSwtichJob = new TargetSwitchJob(target);
+			targetSwtichJob.schedule();
+			try
 			{
-				ILaunchBarManager launchBarManager = UIPlugin.getService(ILaunchBarManager.class);
-				TargetSwitchJob targetSwtichJob = new TargetSwitchJob(target);
-				targetSwtichJob.schedule();
-				try
+				ILaunchDescriptor desc = launchBarManager.getActiveLaunchDescriptor();
+				if (findAppropriateDebugConfig(desc) == null)
 				{
-					ILaunchDescriptor desc = launchBarManager.getActiveLaunchDescriptor();
 					createDefaultDebugConfig();
 					launchBarManager.setActiveLaunchDescriptor(desc);
 				}
-				catch (CoreException e)
-				{
-					Logger.log(e);
-				}
-
+			}
+			catch (CoreException e)
+			{
+				Logger.log(e);
 			}
 		});
 		return performFinish;
 	}
 
+	private ILaunchConfiguration findAppropriateDebugConfig(ILaunchDescriptor descriptor) throws CoreException
+	{
+		ILaunchManager launchManager = DebugPlugin.getDefault().getLaunchManager();
+		IProject project = descriptor.getAdapter(IProject.class);
+		for (ILaunchConfiguration config : launchManager.getLaunchConfigurations())
+		{
+			IResource[] mappedResource = config.getMappedResources();
+			if (mappedResource != null && mappedResource.length > 0 && mappedResource[0].getProject().equals(project)
+					&& config.getType().getIdentifier().contentEquals(IDFLaunchConstants.DEBUG_LAUNCH_CONFIG_TYPE))
+			{
+				return config;
+			}
+		}
+		return null;
+
+	}
+
```

Comment:
Can we somehow make this method and the one in the other class [here](https://github.com/espressif/idf-eclipse-plugin/blob/ea6270ed72c49f80bf8f6f34ff11af1ed4b0c737/bundles/com.espressif.idf.launch.serial.core/src/com/espressif/idf/launch/serial/core/IDFCoreLaunchConfigProvider.java#L102) into some one utility method to reuse the code

Reply:
yes, I was planning to do that and add test coverage for that as well